harness/determined/_generic/_distributed.py

```diff
@@ -34,6 +35,7 @@ def __init__(
         local_size: int,
         cross_rank: int,
         cross_size: int,
+            backend: Optional[str],
```

Comment:
I don't think this is the right way to pass the backend information around.  The backend is 'training loop-specific information', while the DistributedContext is more like a 'training-loop-independent way to ensure a minimum set of distributed communication primitives'

In the case of a totally custom training loop, the backend would just be some random string that the user sets and knows how to interpret, and that's a strange thing to attach to this object I think.